addons/io_hubs_addon/third_party/recast.py

```diff
@@ -0,0 +1,548 @@
+    # for i in range(ntris):
+    #    i1 = dmesh_holder.dmesh.contents.tris[i*3+0]
+    #    i2 = dmesh_holder.dmesh.contents.tris[i*3+1]
+    #    i3 = dmesh_holder.dmesh.contents.tris[i*3+2]
+    #    print("face[%i] = (%i, %i, %i)" % (i, i1, i2, i3))
+    #    bm.faces.new((bm.verts[i1], bm.verts[i2], bm.verts[i3]))  # add a new vert
+
+    bmesh.ops.remove_doubles(bm, verts=bm.verts, dist=0.00001)
+
+    # make the bmesh the object's mesh
+    bm.to_mesh(mesh)
+    bm.free()  # always do this when finished
+
+    # Assign nav mesh color
+    mat = bpy.data.materials.get("Navmesh Material")
+    if mat is None:
+        mat = bpy.data.materials.new(name="Navmesh Material")
+        mat.diffuse_color = bpy.context.scene.recast_navmesh.color
+
+    if mesh.materials:
+        mesh.materials[0] = mat
+    else:
+        mesh.materials.append(mat)
+
+
+class ReacastNavmeshGenerateOperator(bpy.types.Operator):
+    bl_idname = "recast.build_navigation_mesh"
+    bl_label = "Build Navigation Mesh"
+    bl_description = "Build navigation mesh using recast."
+    bl_options = {'REGISTER', 'UNDO'}
+
+    def execute(self, context):
+        # bpy.ops.wm.call_menu(name="ADDITIVE_ANIMATION_insert_keyframe_menu")
+
+        from ..components.definitions.nav_mesh import NavMesh
+        nav_mesh_id = NavMesh.get_name()
+        for ob in bpy.context.selected_objects:
+            if has_component(ob, nav_mesh_id):
+                self.report({'ERROR'}, 'A Navmesh cannot be part of the selection')
+                return {'FINISHED'}
+
+        navMesh = None
+        navMeshes = get_objects_with_component(nav_mesh_id)
+        if navMeshes:
+            navMesh = navMeshes[0]
+
+        addon_prefs = get_addon_pref(context)
+        libpath = os.path.abspath(addon_prefs.recast_lib_path)
+        libpathr = libpath.replace("\\", "/")
+        if not os.path.exists(libpathr):
+            self.report({'ERROR'}, 'File not exists: %s\n' % libpathr)
+            return {'FINISHED'}
+
+        verts, tris = extractTriangulatedInputMesh()
+        vertsCount = len(verts)
+        trisCount = len(tris)
+        nverts = (int)(len(verts) / 3)
+        ntris = (int)(len(tris) / 3)
+        recastData = recastDataFromBlender(bpy.context.scene)
+
+        prevWorkingDir = os.getcwd()
+        nextWorkingDir = os.path.dirname(libpathr)
+        os.chdir(nextWorkingDir)
+        try:
+            recast = ctypes.CDLL(libpathr)
+        except OSError as e:
+            tracebackStr = traceback.format_exc()
+            self.report(
+                {'ERROR'},
+                'Failed to load shared library: %s\nPath to shared library: %s\n\nTraceback: %s' %
+                (str(e),
+                 libpathr, tracebackStr))
+            os.chdir(prevWorkingDir)
+            return {'FINISHED'}
+
+        os.chdir(prevWorkingDir)
+        print("nverts %i" % nverts)
+        print("ntris %i" % ntris)
+        print("cell size %f" % recastData.cellsize)
```

Comment:
I think stuff like this isn't usually printed to the terminal, so should these print statements should be removed?